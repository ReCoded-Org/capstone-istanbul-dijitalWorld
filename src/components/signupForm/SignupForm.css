.signupLogo {
  width: 10rem;
}

.signupContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
}

.signupHeader {
  color: #6c567b;
}

.signupBtn {
  background-color: #6c567b;
  color: white;
  font-weight: 600;
  cursor: pointer;
  width: 10.3125rem;
  height: 3rem;
  font-size: 1.375rem;
  border-radius: 2.1875rem;
  border-style: hidden;
  box-shadow: 0px 11px 24px -7px #b3b3b3;
}

.loginLink {
  display: inline;
  color: #000000;
}
