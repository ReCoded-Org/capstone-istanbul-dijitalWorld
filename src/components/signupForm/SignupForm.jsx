import React, { useState } from 'react';
import { Image, Form, Col, Button, Container, Alert } from 'react-bootstrap';
import { Redirect } from 'react-router-dom';
import { auth } from '../../firebase';
import Logo from '../../images/www-logo.png';
import './SignupForm.css';

const SignupForm = () => {
  // This state manages the redirecting to the home page that happens after a successful signup
  const [redirect, setRedirect] = useState(false);
  // Alert is null if not present. If present, has two keys, message and status.
  // Status's value will decide the appearance of the alert
  const [alert, setAlert] = useState(null);
  const [signupInfo, setSignupInfo] = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    repeatPassword: '',
  });

  // This function uses firebase auth to register the user
  // In case of any error it will change the alert state to render a component that shows an error message to the user
  // If the login was successful it will show  a success alert message and then redirect the user to the home page
  const handleSignup = async (e) => {
    e.preventDefault();
    if (signupInfo.password !== signupInfo.repeatPassword) {
      setAlert({ message: "Passwords doesn't match", status: 'danger' });
      return;
    }
    try {
      const response = await auth.createUserWithEmailAndPassword(
        signupInfo.email,
        signupInfo.password,
      );
      response.user.sendEmailVerification();
      response.user.updateProfile({
        displayName: signupInfo.firstName + " " + signupInfo.lastName,
      })
    } catch (error) {
      setAlert({ message: error.message, status: 'danger' });
      return;
    }
    setSignupInfo({
      firstName: '',
      lastName: '',
      email: '',
      password: '',
      repeatPassword: '',
    });
    setAlert({ message: 'Your registration is completed!', status: 'success' });
    setTimeout(() => setRedirect(true), 1000);
  };

  const handleChange = (key) => (e) => {
    setSignupInfo({ ...signupInfo, [key]: e.target.value });
  };

  return (
    <Container className="signupContainer">
      <Image src={Logo} className="signupLogo" />
      <Form onSubmit={handleSignup} className="form">
        <Form.Row>
          <Form.Group as={Col} controlId="formGridFirstName" className="mt-3">
            <Form.Control
              required
              type="name"
              placeholder="First name"
              value={signupInfo.firstName}
              onChange={handleChange('firstName')}
            />
          </Form.Group>

          <Form.Group as={Col} controlId="formGridLastName" className="mt-3">
            <Form.Control
              required
              type="name"
              placeholder="Last name"
              value={signupInfo.lastName}
              onChange={handleChange('lastName')}
            />
          </Form.Group>
        </Form.Row>

        <Form.Group controlId="formGridEmail" className="mt-3">
          <Form.Control
            type="email"
            placeholder="Your email"
            value={signupInfo.email}
            onChange={handleChange('email')}
          />
        </Form.Group>

        <Form.Row>
          <Form.Group as={Col} controlId="formGridPassword" className="mt-3">
            <Form.Control
              required
              type="password"
              placeholder="Password"
              value={signupInfo.password}
              onChange={handleChange('password')}
            />
          </Form.Group>

          <Form.Group as={Col} controlId="formGridRepeatPassword" className="mt-3">
            <Form.Control
              required
              type="password"
              placeholder="Repeat password"
              value={signupInfo.repeatPassword}
              onChange={handleChange('repeatPassword')}
            />
          </Form.Group>
        </Form.Row>

        <Button variant="primary" type="submit" className="signupBtn mt-3">
          Sign Up
        </Button>
      </Form>
      {alert && (
        <Alert className="mt-3" variant={alert.status} onClose={() => setAlert(null)} dismissible>
          <p>{alert.message}</p>
        </Alert>
      )}
      {redirect && <Redirect to="/" />}
      <p className="mt-3">
        Already Have An Account ?{' '}
        <a href="/login" className="loginLink">
          Login
        </a>
      </p>
    </Container>
  );
};

export default SignupForm;
