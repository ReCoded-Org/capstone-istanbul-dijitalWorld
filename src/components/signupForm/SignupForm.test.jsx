import React from 'react';
import renderer from 'react-test-renderer';
import SignupForm from './SignupForm';

it('renders correctly', () => {
  const tree = renderer.create(<SignupForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
