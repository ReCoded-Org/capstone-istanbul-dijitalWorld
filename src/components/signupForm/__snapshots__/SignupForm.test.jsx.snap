// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="signupContainer container"
>
  <img
    className="logo"
    src="www-logo.png"
  />
  <h2
    className="signupHeader"
  >
    Signup Now
  </h2>
  <form
    className=""
  >
    <div
      className="form-row"
    >
      <div
        className="mt-3 form-group col"
      >
        <input
          className="form-control"
          id="formGridFirstName"
          placeholder="FirstName"
          type="name"
        />
      </div>
      <div
        className="mt-3 form-group col"
      >
        <input
          className="form-control"
          id="formGridLastName"
          placeholder="LastName"
          type="name"
        />
      </div>
    </div>
    <div
      className="mt-3 form-group"
    >
      <input
        className="form-control"
        id="formGridEmail"
        placeholder="Your Email"
        type="email"
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="mt-3 form-group col"
      >
        <input
          className="form-control"
          id="formGridPassword"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="mt-3 form-group col"
      >
        <input
          className="form-control"
          id="formGridRepeatPassword"
          placeholder="Repeat Password"
          type="password"
        />
      </div>
    </div>
    <button
      className="signupBtn mt-3 btn btn-primary"
      disabled={false}
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <p
    className="mt-3"
  >
    Already Have An Account ?
     
    <a
      className="loginLink"
      href="/login"
    >
      Login
    </a>
  </p>
</div>
`;
